src/maptm/src/maptm_ovsdb.c

```diff
@@ -148,21 +148,20 @@ bool maptm_persistent(void)
     }
 
     // Get persistent storage value
-    size = osp_ps_get(ps, "MAPT_SUPPORT", mapt_support, strlen(mapt_support));
+    size = osp_ps_get(ps, "MAPT_SUPPORT", mapt_support, sizeof(mapt_support));
     if (size <= 0)
     {
         LOGE("Failed when getting persistent storage value");
-        goto out;
+        snprintf(mapt_support, sizeof(mapt_support), "%s", "true");
+        
+        // Set persistent storage value
+        if (!(osp_ps_set(ps, "MAPT_SUPPORT", mapt_support, sizeof(mapt_support))))
+        {
+            LOGE("Failed when setting persistent storage value");
+            goto out;
+        }
```

Comment:
```suggestion
        if (size < 0) {
            LOGE("Failed when getting persistent storage value, will try to set it anyway");
        } else {  // (size == 0)
            LOGI("Persistent storage value does not exist, setting the default (true)");
        }
 
        STRSCPY(mapt_support, "true");

        // Set persistent storage value
        if (osp_ps_set(ps, "MAPT_SUPPORT", mapt_support, strlen(mapt_support)+1) <= 0)
        {
            LOGE("Failed when setting persistent storage value");
            goto out;
        }
```

Always use stlen()+1 when writing to the persistent storage. This guarantees that the text will be null terminated when reading it back. OTOH, using sizeof() when writing is wasteful (and of course confusing for the reader).